src/library.js

```diff
@@ -2388,6 +2387,40 @@ mergeInto(LibraryManager.library, {
   _emscripten_get_now_is_monotonic__deps: ['$nowIsMonotonic'],
   _emscripten_get_now_is_monotonic: () => nowIsMonotonic,
 
+  __emscripten_atomics_sleep__internal: true,
+  // In a browser without cross origin isolation, SharedArrayBuffer is deleted
+  // from the global scope:
+  // https://html.spec.whatwg.org/multipage/webappapis.html#creating-a-new-javascript-realm
+  // However, it says:
+  //
+  // > Web developers can still get at the constructor through
+  // > `new WebAssembly.Memory({ shared:true, initial:0, maximum:0}).buffer.constructor`.
+  //
+  // That is what we are up to here. Use an IIFE to avoid shadowing
+  // SharedArrayBuffer globally.
+  __emscripten_atomics_sleep__postset: `
+    var waitBuffer, canBlock;
```

Comment:
Ideally `canBlock` should be a wasm variable and then we can access it from C without calling into JS.

Reply:
BTW, for historical reasons we do not set `_emscripten_thread_supports_atomics_wait` to true to the main node thread.  I suggest we keep that the same for the initial version of this change and then see if we can make it true for the node main thread as a followup.